dali/python/nvidia/dali/pipeline.py

```diff
@@ -769,7 +769,14 @@ def feed_input(self, data_node, data, layout = None, cuda_stream = None, use_cop
             _data_node._check(data_node)
             name = data_node.name
 
-        from nvidia.dali.external_source import _check_data_batch
+        from nvidia.dali.external_source import _check_data_batch, _ExternalSourceGroup
+
+        # Check if user uses feed_input on an external_source operator that was initialized with 'source'
+        if next((op._callback is not None for op in self._ops if op.name == name), False):
+            _, _, _, values = inspect.getargvalues(inspect.currentframe().f_back)
```

Comment:
One more thing crossed my mind that maybe instead of inspecting the call stack we could somehow modify this interface, for instance that there is "private" _feed_input and public feed_input with the check, and external source while poked for more data by the Pipeline can receive some feed callback, future like object or sth that does not perform this check. Or just call the private _feed_input. I don't have strong opinion on that, just food for thought.

@mzient What do you think?

Reply:
That was my first instinct as well, but I wanted to modify the code as little as possible without adding more layers of abstraction.

Reply:
But maybe it's not that big of a change and better in the long run.